### PR TITLE
PBI-205: Create vendor evaluation

### PR DESCRIPTION
**Pull request by valenciusap17**
## Describe your changes
- Add new table storing evaluations for a vendor
- Create an API to create an evaluation
- Implemented service and accessor to create vendor evaluation
- Screenshot:
image

## Issue ticket number and link
- PBI-205: https://linear.app/adudu/issue/ADU-205/backend-form-evaluasi

### CODE REVIEW

File: internal/vendors/service.go [hunk 1]
```diff
@@ -82,6 +83,15 @@ func (v *VendorService) AutomatedEmailBlast(ctx context.Context, productName str
 	return v.executeBlastEmail(ctx, vendors, *template)
 }
 
+func (v *VendorService) CreateEvaluation(ctx context.Context, evaluation *VendorEvaluation) (*VendorEvaluation, error) {
+	id, _ := helper.GenerateRandomID()
```
Comment (edutjie) [inline internal/vendors/service.go:87]: if didn't you handle the error?

File: internal/vendors/accessor.go [hunk 2]
```diff
@@ -362,6 +368,17 @@ func (p *postgresVendorAccessor) BulkGetByProductName(_ context.Context, product
 	return vendors, nil
 }
 
+func (p *postgresVendorAccessor) CreateEvaluation(ctx context.Context, evaluation *VendorEvaluation) (*VendorEvaluation, error) {
+	query := createEvaluation
```
Comment (valeelim) [inline internal/vendors/accessor.go:372]: do we need this variable?

File: router/vendor.go [hunk 3]
```diff
@@ -170,4 +172,30 @@ func NewVendorEngine(
 			"message": "Emails successfully sent",
 		})
 	})
+	r.POST(cfg.Evaluation, func(ctx *gin.Context) {
+		utils.Logger.Info("Received vendor evaluation request")
+
+		vendorEvaluation := &vendors.VendorEvaluation{}
```
Comment (valeelim) [inline router/vendor.go:178]: this should be created in the service layer IMO

File: internal/vendors/service.go (hunk 1 above)
Comment (valenciusap17) [reply internal/vendors/service.go:87]: because the error returned by rand.Int is a rare case, so I ignore the error

File: router/vendor.go (hunk 3 above)
Comment (valeelim) [reply router/vendor.go:178]: nvmd I didn't see quite correctly, forgive me 😭

**Review by valeelim — APPROVED** [review]
Supercalifragilisticexpialidocious job 👌

**Pull request merged**